pytket/binders/pauli.cpp

```diff
@@ -237,6 +237,184 @@ PYBIND11_MODULE(pauli, m) {
           "The list of Pauli terms")
       .def("__eq__", &PauliStabiliser::operator==)
       .def("__ne__", &PauliStabiliser::operator!=);
+
+  py::class_<QubitPauliTensor>(
+      m, "QubitPauliTensor",
+      "A tensor formed by Pauli terms, consisting of a sparse map from "
+      ":py:class:`Qubit` to :py:class:`Pauli` (implemented as a "
+      ":py:class:`QubitPauliString`) and a complex coefficient.")
+      .def(
+          py::init<Complex>(),
+          "Constructs an empty QubitPauliTensor, representing the identity.",
+          py::arg("coeff") = 1.)
+      .def(
+          py::init<Qubit, Pauli, Complex>(),
+          "Constructs a QubitPauliTensor with a single Pauli term.",
+          py::arg("qubit"), py::arg("pauli"), py::arg("coeff") = 1.)
+      .def(
+          py::init([](const std::list<Qubit> &qubits,
+                      const std::list<Pauli> &paulis, const Complex &coeff) {
+            return QubitPauliTensor(QubitPauliString(qubits, paulis), coeff);
+          }),
+          "Constructs a QubitPauliTensor from two matching lists of "
+          "Qubits and Paulis.",
+          py::arg("qubits"), py::arg("paulis"), py::arg("coeff") = 1.)
+      .def(
+          py::init<QubitPauliMap, Complex>(),
```

Comment:
How is a `QubitPauliMap` defined?

I think I'm right in saying that this class wasn't exposed to python either. I find references to it in the `pytket.pauli` docs but can't seem to find a definition.

Reply:
It is the C++ typedef for a map from Qubit to Pauli. pybind11 translates it to dict, hence why the typing in the docs displays it as `Dict[pytket.circuit.Qubit, pytket.pauli.Pauli]`. I will change the docstring to remove this name and prevent confusion